app.js

```diff
@@ -6,95 +6,112 @@ var express = require('express');
 var compression = require('compression');
 var bodyParser = require('body-parser');
 var multer = require('multer');
-var pkg_info = require('./package.json');
 var fs = BBPromise.promisifyAll(require('fs'));
 
 
 /**
- * The main application object
+ * Promise create an express app and initialize it
+ * @param options
+ * @returns {bluebird}
  */
-var app = express();
+function initApp(options) {
+    return new BBPromise(function (resolve) {
+        // The main application object
+        var app = express();
 
-/* Basic configuration */
-
-// disable the X-Powered-By header
-app.set('x-powered-by', false);
-// disable the ETag header - users should provide them!
-app.set('etag', false);
-// enable compression
-app.use(compression({level: 3}));
-// use the JSON body parser
-app.use(bodyParser.json());
-// use the application/x-www-form-urlencoded parser
-app.use(bodyParser.urlencoded({ extended: true }));
-// use the multipart/form-data
-app.use(multer());
-// serve static files from static/
-app.use('/static', express.static(__dirname + '/static'));
-
-/*** More configuration of app comes here ***/
-
-
-function loadRoutes () {
+        // get the options and make them available in the app
+        app.logger = options.logger;    // the logging device
+        app.metrics = options.metrics;  // the metrics
+        app.conf = options.config;      // this app's config options
+        app.info = require('./package.json'); // this app's package info
 
-    // get the list of files in routes/
-    return fs.readdirAsync(__dirname + '/routes')
-    .map(function(fname) {
-        // ... and then load each route
-        // but only if it's a js file
-        if(!/\.js$/.test(fname)) {
-            return;
+        // ensure some sane defaults
+        if (!app.conf.port) {
+            app.conf.port = 8888;
         }
-        // import the route file
-        var route = require(__dirname + '/routes/' + fname);
-        route = route(app);
-        // check that the route exports the object we need
-        if(route.constructor !== Object || !route.path || !route.router) {
-            throw new Error('routes/' + fname + ' does not export the correct object!');
+        if (!app.conf.interface) {
+            app.conf.interface = '0.0.0.0';
         }
-        // all good, use that route
-        app.use(route.path, route.router);
-    }).then(function() {
-        // route loading is now complete, return the app object
-        return app;
+
+        // disable the X-Powered-By header
+        app.set('x-powered-by', false);
+        // disable the ETag header - users should provide them!
+        app.set('etag', false);
+        // enable compression
+        app.use(compression(app.conf.compressionLevel ? {level: app.conf.compressionLevel} : {}));
```

Comment:
I will switch the default to 3, but I feel it would be a good finetuning item once deployed.